### DISCUSSION TITLE
Parsing raw HTML? i.e converting HTML to PDF

### DISCUSSION THREAD

**Discussion by ghost** [rubycdp/ferrum#456] [Q&A]
Hi,

Is there a way with Ferrum to parse raw HTML that will also load CSS/JS instead of visiting a URL?

A workaround would be to save the file and navigate to it `file://....` but just asking in case it's possible.

Thanks

**Comment by route** [comment]
I think you are looking for https://github.com/rubycdp/ferrum/commit/13ab43d4983d5763f0f7dd0a647289d3561778a6#diff-593bb59ec2e71055f0e04ace86070028R239-R243

**Reply by ghost** [reply]
I'm pretty sure I tested this and wasn't able to replace anything else but the body. The head wouldn't follow. I would need to re-test to reconfirm

**Reply by route** [reply]
I do believe it works.
